### PR TITLE
Export use_directory() and wrote documentation for it

### PR DESCRIPTION
**Pull request by s-fleck**

### CODE REVIEW

File: R/helpers.R [hunk 1]
```diff
@@ -99,6 +99,21 @@ use_dependency <- function(package, type, base_path = ".") {
...
 
+#' Use a directory.
+#'
+#' `use_directory()` creates a directory (if it does not already exist) in the
+#' package root dir. This function powers many of the other `use_` functions
+#' such as [use_data()] and [use_vignette()].
+#'
+#' @return `TRUE` (invisibly) if the directory was successfully created.
+#'
+#' @inheritParams use_template
```
Comment (hadley) [inline R/helpers.R]: Also need to document path

File: R/helpers.R [hunk 2]
```diff
@@ -99,6 +99,21 @@ use_dependency <- function(package, type, base_path = ".") {
   invisible()
 }
 
+#' Use a directory.
+#'
+#' `use_directory()` creates a directory (if it does not already exist) in the
+#' package root dir. This function powers many of the other `use_` functions
+#' such as [use_data()] and [use_vignette()].
+#'
+#' @return `TRUE` (invisibly) if the directory was successfully created.
```
Comment (hadley) [inline R/helpers.R]: I think you can leave this off. I'd rather do it for all function in one swoop

**Comment by s-fleck** [comment]
Hmm im kinda new to this. To me it shows two commits for the pull request now: the initial one, and the one that adds documentation for path and adds the NEWS.md bullet point.

**Comment by hadley** [comment]
Hmmm, now it looks mostly ok. I'll finish off the process. Thanks!

**Comment by s-fleck** [comment]
Ok thanks, no idea what went wrong on the initial push

**Pull request merged**